tools/generate_tables_sql_and_testNames.php

```diff
@@ -1,118 +1,224 @@
+            }
+        }
+        $output .= "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB"
+            ." DEFAULT CHARSET=utf8;\n";
+        $fp      = fopen($file_name, "w");
+        fwrite($fp, $output);
+        // Check if Table Name exists
+        if (empty($sql_query_table_name)) {
+            $error_message = "Instrument(s) file has invalid instrument data.";
+        } else {
+            fwrite(
+                $fp,
+                "REPLACE INTO test_names (Test_name, Full_name, Sub_group) "
+                ."VALUES ('$sql_query_table_name', '$title', 1);\n"
+            );
+            // Writes a record of the changes to a file but is not sourced directly.
+            foreach ($pages as $page_number => $page) {
+                fwrite(
+                    $fp,
+                    "INSERT INTO instrument_subtests "
+                    ."(Test_name, Subtest_name, Description)"
+                    ." VALUES ('$sql_query_table_name', '"
+                    .$sql_query_table_name."_page".($page_number+1)
+                    ."', '$page');\n"
+                );
+            }
         }
-
+        // Writes the error message to the sql file.
+        if (!empty($error_message)) {
+            fwrite(
+                $fp,
+                "/* The SQL file is invalid! \n"
+                ."tools/generate_tables_sql_and_testNames.php error_message:\n"
+                ."$error_message\n"
+                ."*/"
+            );
+        }
+        fclose($fp);
     }
-    $output.="PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;\n";
-    print "Filename: $filename\n";
-    $fp=fopen($filename, "w");
-    fwrite($fp, $output);
-    fwrite($fp, "REPLACE INTO test_names (Test_name, Full_name, Sub_group) VALUES ('$tablename', '$title', 1);\n");
-    foreach($pages as $pageNo => $page) {
-        fwrite($fp, "INSERT INTO instrument_subtests (Test_name, Subtest_name, Description) VALUES ('$tablename', '" . $tablename  . "_page" . ($pageNo+1) . "', '$page');\n");
-
+    // Execute the SQL tasks if no error_message.
+    if ($db->isConnected() && empty($error_message)) {
+        if (!$db->tableExists($sql_query_table_name)) {
+            // Table doesn't exist!
+            $sql_query_statement .= implode($sql_query_columns);
+            $sql_query_statement .=
+                "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB "
+                ."DEFAULT CHARSET=utf8;\n";
+            $statement            = $db->prepare($sql_query_statement);
+            $statement->execute();
+        } else {
+            $error_message = "Instrument(s) file uses existing name.";
+        }
+        $statement = $db->prepare(
+            'REPLACE INTO test_names (Test_name, Full_name, Sub_group)'
+            .' VALUES (?, ?, 1)'
+        );
+        $statement->bindParam(1, $sql_query_table_name, PDO::PARAM_STR);
+        $statement->bindParam(2, $title, PDO::PARAM_STR);
+        if ($execute_mysql_query) {
```

Comment:
I would move this test (and the one on line 193) to line 174 ( or maybe 163) as there is no need to create all these statement if they will never be executed